### PR TITLE
Fixed enhanced gas fee bug

### LINKED ISSUE #13800: [Bug]: Enhanced Gas Fee UI saves the same values regardless of network
### Describe the bug

Using the Enhanced Gas Fee UI and ticking the "Always use these values and advanced setting as default." option, the max base fee and priority fee values are saved regardless of blockchain used.

Expectation: values should be saved per blockchain, as different blockchains have widely different gas fees.

### Steps to reproduce

1. Start a 0 ETH transaction to your own address on Ethereum
2. Edit Max Base Fee and Priority Fee. Tick "Always use these values and advanced setting as default."
3. Reject the transaction
4. Switch to another blockchain (for example, Polygon)
5. Start a 0 ETH transaction to your own address
6. Edit gas fee. The gas values will be the same you saved on Ethereum

### Error messages or log output

_No response_

### Version

10.10.1

### Build type

_No response_

### Browser

Firefox

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by filipsekulic**
Fixes: https://github.com/MetaMask/metamask-extension/issues/13800

Screencast: 
https://user-images.githubusercontent.com/92310504/160845177-72378956-1a64-42c2-9a07-60d1f46c310a.mov

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

**Comment by danjm** [comment]
I think this looks pretty good, but I am realizing we need to add a migration. I have set some time on the calendar for next week to introduce that.

**Review by danjm — CHANGES_REQUESTED** [review]
wait until we add a data migration to merge this

**Comment by filipsekulic** [comment]
> @filipsekulic gonna need a rebase and I assume bump your migration number to avoid the conflict on 073

Done :white_check_mark:

**Comment by brad-decker** [comment]
@danjm you reviewed this previously could you possibly take first pass at this now that it has a migration?

File: app/scripts/controllers/transactions/index.test.js
Comment (jpuri) [inline app/scripts/controllers/transactions/index.test.js:805]: Not sure why is thus stud needed now since it was required before.

**Review by jpuri — DISMISSED** [review]
👍 Code changes look good, just a small feedback.

**Comment by jpuri** [comment]
It might be useful to change this message also to reflect the network: @filipsekulic

**Comment by filipsekulic** [comment]
> 

Fixed:
Screenshot 2022-11-25 at 12 07 16

File: app/scripts/controllers/transactions/index.test.js
Comment (filipsekulic) [reply app/scripts/controllers/transactions/index.test.js:805]: This PR was created long time ago...

Comment (jpuri) [reply app/scripts/controllers/transactions/index.test.js:805]: Are these still needed now ?

Comment (filipsekulic) [reply app/scripts/controllers/transactions/index.test.js:805]: I suppose so, as it can also be seen on the main branch, within the same file.

Comment (jpuri) [reply app/scripts/controllers/transactions/index.test.js:805]: `stub3` look new to this PR.

Comment (filipsekulic) [reply app/scripts/controllers/transactions/index.test.js:805]: Oh, I thought you were talking about the `maxPriorityFeePerGas: TEST_MAX_PRIORITY_FEE_PER_GAS`...
So, here is the thing. This PR introduces saving the `maxFeePerGas` and `maxPriorityFeePerGas` for every network separately. Because of it, those fees will look like this: 
`      "0x5": {
        "maxBaseFee": "75",
        "priorityFee": "2"
      }`
Are we on the same page now, I mean does this make sense to you or are you suggesting something else?

**Review by jpuri — DISMISSED** [review]
LGTM

File: app/scripts/migrations/078.test.js
Comment (brad-decker) [inline app/scripts/migrations/078.test.js:4]: We typically like to add some tests here to ensure that nothing changes unexpectedly, like in the case where advancedGasFee isn't set for some reason, or make sure unaffected keys do not get changed in anyway. Please add a few more test cases even they feel redundant, use the other tests for examples

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: app/scripts/migrations/078.test.js
Comment (filipsekulic) [reply app/scripts/migrations/078.test.js:4]: Got it!
Added here: 8677314c1999ffa3995c3cb693e64a8d9841009d

**Review by jpuri — DISMISSED** [review]

**Review by jpuri — DISMISSED** [review]

**Comment by bschorchit** [comment]
@danjm @brad-decker pinging for a re-review in this PR as you have requested changes in the past

**Review by brad-decker — APPROVED** [review]

**Comment by bschorchit** [comment]
Closing in favor of https://github.com/MetaMask/metamask-extension/pull/20576

**Pull request closed without merging**
